src/lua/api-gateway/aws/AWSIAMCredentials.lua

```diff
@@ -189,4 +208,31 @@ function AWSIAMCredentials:getSecurityCredentials()
     return cache.AccessKeyId, cache.SecretAccessKey, cache.Token, cache.ExpireAt, cache.ExpireAtTimestamp
 end
 
+-- Retrieves a IMDSv2 Token if it hasn't been done yet or if the previous token has expired
+function AWSIAMCredentials:retrieveIMDSv2Token()
+
+    if (cache.ExpireIMDSv2AtTimestamp == nil or cache.ExpireIMDSv2AtTimestamp - os.time() <= 0 ) then
```

Comment:
Should this have a little fudge factor so the token doesn't expire right after this check? Maybe if it expires in 10 seconds?